docs/localization/LOCALIZATION.md

```diff
@@ -74,343 +78,465 @@ export function localizationConfigurationFactory() {
+#### How to generate a component bundle
+
+```shell
+ng run project-name:i18n
+```
+
+It will create an `i18n` folder in the component folder, with an `en-GB.json` file containing the mapping between the localization key and the associated default value.
+
+Check the list of options available by running:
+```shell
+ng run project-name:i18n --help
+```
+
+#### How to generate an application bundle
+
+```shell
+ng run project-name:generate-translations
+```
+It will create the application bundle for each supported language.
+
+Check the list of options available by running:
+
+```shell
+ng run project-name:generate-translations --help
+```
+
+## Additional features
+
+### How to localize a date, decimal and currency
+
+Use Angular's built-in [DatePipe](https://angular.io/api/common/DatePipe), [DecimalPipe](https://angular.io/api/common/DecimalPipe) and [CurrencyPipe](https://angular.io/api/common/CurrencyPipe) pass the current locale as the last parameter.
+The locale is read from `this.localizationService.getCurrentLanguage()`. To be able to use this, your component must inject the `LocalizationService` dependency as a parameter of the constructor.
 
 ```typescript
-import {Component, Input} from  '@angular/core';
-import {Localization, LocalizationService, Translatable} from  '@o3r/localization';
-import {SimpleHeaderPresTranslation, translations} from  './simple-header-pres.translation';
+import { Component, Input } from '@angular/core';
+import { Localization, LocalizationService, Translatable } from '@o3r/localization';
+import { SimpleHeaderPresTranslation, translations } from './simple-header-pres.translation';
 
 @Component({
   selector: 'o3r-simple-header-pres',
   styleUrls: ['./simple-header-pres.style.scss'],
   templateUrl: './simple-header-pres.template.html'
 })
+export class SimpleHeaderPresComponent implements Translatable<SimpleHeaderPresTranslation>, ... {
 
-export  class SimpleHeaderPresComponent implements Translatable<SimpleHeaderPresTranslation>, ... {
-  /**
-  * Localization of the component
-  */
-  @Input()
-  @Localization('./simple-header-pres.localization.json')
-  public translations: SimpleHeaderPresTranslation;
-
+  // ...
 
-  constructor(
-    public localizationService: LocalizationService) {
-    this.translations = translations;
+  constructor(public localizationService: LocalizationService) {
+    // ...
   }
+
   /**
    * Called upon language change to set current language
    * @param  language
    */
-  useLanguage(language: string) {
+  public useLanguage(language: string) {
     this.localizationService.useLanguage(language);
   }
 }
 ```
 
 Now we can start using pipes:
 
-```typescript
-
-<p>{{today | date:'fullDate': '': localizationService.getCurrentLanguage()}}</p>
-
-<p>{{1.5487 | number: '': localizationService.getCurrentLanguage()}}</p>
+```html
+<p>{{today | date: 'fullDate' : '' : localizationService.getCurrentLanguage()}}</p>
```

Comment:
not from this PR, but it is not a good idea to call methods in the template. You can create a public property 'currentLanguage' and initialize it in the constructor with 'localizationService.getCurrentLanguage()' and then use the property in template.